lib/services/database.go

```diff
@@ -213,6 +213,10 @@ func NewDatabasesFromRDSClusterCustomEndpoints(cluster *rds.DBCluster) (types.Da
 
 // NewDatabaseFromRedshiftCluster creates a database resource from a Redshift cluster.
 func NewDatabaseFromRedshiftCluster(cluster *redshift.Cluster) (types.Database, error) {
+	if cluster.Endpoint == nil {
```

Comment:
Perhaps we should add a note saying that this can happen in the course of normal operation due to the race condition described in #10592 and thus is not an immediate cause of worry.

Reply:
Also, could we filter these earlier, during `(*redshiftFetcher).Get` maybe?